src/TesApi.Web/Scheduler.cs

```diff
@@ -125,14 +126,24 @@ private async ValueTask OrchestrateTesTasksOnBatch(CancellationToken stoppingTok
 
             var startTime = DateTime.UtcNow;
 
-            foreach (var tesTask in tesTasks)
+            await Task.WhenAll(await batchScheduler.ProcessTesTasksAsync(tesTasks, stoppingToken).Select(ProcessOrchestrationResult).ToArrayAsync(stoppingToken));
```

Comment:
how many tasks is this potentially awaiting?  Would it make sense to await batches of 100 of them at a time or something?  Just want to be sure this doesn't await 5000 tasks that causes massive memory load

Reply:
However many of them there are. The GC is most efficient (and effective) when you load up a bunch of objects and then drop them all, and this basically does that.

Dividing the list into batches by size makes the code more complicated, and actually makes the job of the GC harder, but would mitigate spikes in the memory usage that could result in OutOfMemory errors. Performance metrics is the place to determine if that's a concern, it's not something we can know at this point.